Source/AppAuthCore/OIDServiceConfiguration.m

```diff
@@ -185,8 +185,8 @@ - (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
     return nil;
   }
 
-  OIDServiceDiscovery *discoveryDocument = [aDecoder decodeObjectOfClass:[OIDServiceDiscovery class]
-                                                                  forKey:kDiscoveryDocumentKey];
+  OIDServiceDiscovery *discoveryDocument = [aDecoder decodeObjectOfClasses:[NSSet setWithObjects:[OIDServiceDiscovery class], [NSDictionary class], [NSArray class], nil]
```

Comment:
The "fix" is weird to me. Now `discoveryDocument` can also be a dictionary or an array.
Won't this down the road throw some "unrecognized selector sent to instance" exceptions because we are expecting the object to be of the class `OIDServiceDiscovery`?

Reply:
> The class of the object may be any class in the classes set, or a subclass of any class in the set.

This is what the Apple docs say. AFAIK this is unrelated to the properties' classes.

> This I have tested and it is working as expected.

But have you tested all paths? What happens when the decoding fails or creates a dictionary? Were you actually able to reproduce the bug?

> Do you have any other proposals?

The core of the issue seems to be nullability. Maybe return `nil` if no `discoveryDocument` can be created?
I'm not familar enough with this to suggest an educated workaround though, so you'd need to check if this would even be the intended behavior. 

That said, [this comment](https://github.com/openid/AppAuth-iOS/issues/479#issuecomment-673553053) is quite interesting. Did you try the suggested fix?

Reply:
> Were you actually able to reproduce the bug?

Yes, I could reproduce by logging in and restarting the sample app in Identity to read and parse the cached value, let me try the other comment you linked.

Reply:
I have managed to fix decoding by force disabling the secure decoding, while still keeping it on the encoding side: https://github.com/sumup/identity-ios/pull/64/commits/e47a9e44eaa9594eacc4fa38d14b6d4248c659f7

Closing this PR, thanks for your input!